enum.c

```diff
@@ -132,21 +132,27 @@ enum_grep0(VALUE obj, VALUE pat, VALUE test)
 }
 
 /*
- *  call-seq:
- *     enum.grep(pattern)                  -> array
- *     enum.grep(pattern) { |obj| block }  -> array
+ * call-seq:
+ *   grep(pattern) -> new_array
+ *   grep(pattern) {|element| ... } -> new_array
+ *
+ * Returns a new \Array of objects either selected by the given +pattern+
```

Comment:
1. Why "new" both in the call-seq and description (this seems to imply that the receiver is an array and the method returns a new, different array)?
2. Why "\Array" and not "Array" (this would produce a helpful link to the Array class)?
3. "either selected by the given pattern or returned by the block" does not properly describe the behavior; in both cases the selection via the given pattern applies

Reply:
1.  The returned object is in fact new, regardless of the class of the receiver.
2.  Take this up over at method_documentation.rdoc.
3. Disagree.  Returned Array is in fact either elements selected by the pattern or objects returned by the block.  The 1-line summary need not specify on what terms the block is called.

Reply:
1\. yes, they are (of course) new, so no need to mention it. See the explanation already given above:

The established practice is like

```
array.join -> string
array.reverse -> new_array
```

In both cases the returned object is newly created, but "new_thing" is usually used when the receiver is a "thing" and the method returns a different "thing" (only to make clear that the method dos not return self).

3\. Disagree, the selection based on the pattern takes place always, the block only maps the selected elements; the new version can be misunderstood in such a way, that, with block given, no selection happens.

Reply:
1.  If I thought establish practice was immutable, I wouldn't be doing these revisions.
3.  If you have a suggested revision for the 1-lines, please put it here.

Reply:
1\. These documentations have been improved and tweaked over the course of years, by many contributors; and there are reasons for the common practices that evolved in this process; so lightheartedly waving everything aside with such an argument seems a bit extreme.

Reply:
As I said,  I have an approval for this PR, and so would like to squash and merge.

Reply:
1. I agree that `array` is better than `new_array` here, since the receiver is not an array.
2. The decision for `Array` vs `\Array` can be made by the person doing the work (so whatever @BurdetteLamar prefers in this case).
3. `either selected by the given pattern or returned by the block` is ambiguous, and the literal interpretation does not accurately describe the behavior, as @stomar mentioned.  I can't think of a better version that is shorter, but we may want to use a more verbose form, such as `selected by the given pattern, and returned by the block if given`.